torchao/float8/benchmarking/float8_training_benchmark.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# This script can be used to launch a torchtitan float8 training run
+# with the given parameters,
+
+# script arguments
+BATCH_SIZE=${BATCH_SIZE:-1}
+STEPS=${STEPS:-100}
+
+# temporary log file which is deleted after performance data is parsed out and metrics are calculated.
+LOG_FILE="/tmp/float8_training_log.txt"
+
+# validate user has specified torchtitan root directory
+if [ -z "${TORCHTITAN_ROOT}" ]; then
+  echo "Error: TORCHTITAN environment variable is not set. Please set it before running this script."
+  echo "Usage: TORCHTITAN_ROOT=<directory> ./float8_training_benchmark.sh"
+  echo "Optional parameters configurable via environment variables:"
+  echo " * FLOAT8_RECIPE: "rowwise" or "tensorwise". if set, use float8 training with the specified recipe. otherwise, use bf16 mixed precision training."
+  echo " * BATCH_SIZE: defaults to 1."
+  echo " * STEPS: defaults to 100."
+  exit 1
+fi
+
+# validate recipe name
+if [ -n "${FLOAT8_RECIPE}" ]; then
+  if [ "$FLOAT8_RECIPE" != "rowwise" ] && [ "$FLOAT8_RECIPE" != "tensorwise" ]; then
```

Comment:
nit: this is already checked twice (once in torchtitan, once in Float8LinearConfig), IMO we don't need to check it a third time :)